### ISSUE TITLE
azurerm_kubernetes_cluster: Support for container health monitoring on creation

### ISSUE THREAD

**Issue by tkeber** [terraform-providers/terraform-provider-azurerm#1290]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When creating a new kubernetes cluster on the azure portal, I have the option to enable Container monitoring and specify a Log Analytics workspace.  
It would be great if that could also be enabled in the `azurerm_kubernetes_cluster` configuration.

### New or Affected Resource(s)

* azurerm_kubernetes_cluster

### Potential Terraform Configuration

```HCl
resource "azurerm_log_analytics_workspace" "test" {
  ...
}

resource "azurerm_kubernetes_cluster" "test" {
  name                = "test"
  ...
  monitoring {
    workspace_id = "${azurerm_log_analytics_workspace.test.id}"
  }
}
```

### References

Similar to:
* https://github.com/terraform-providers/terraform-provider-azurerm/issues/1276

**Comment by tombuildsstuff** [comment]
hi @tkeber 

Thanks for opening this issue :)

When you enable monitoring for an AKS Cluster this is actually deploying an OMS Solution [which is already available in Terraform](https://www.terraform.io/docs/providers/azurerm/r/log_analytics_solution.html) (for example, if you create an AKS cluster and enable this in the portal, viewing the "automation script" will show the OMS solution that's being deployed). Since support for this is already possible in Terraform - I'm going to close this issue for the moment.

Thanks!

**Comment by tkeber** [comment]
Hi @tombuildsstuff,

thanks for the quick response! I probably haven't expressed my issue properly, so let me rephrase it:

I have an existing Log Analytics workspace (that was created with `azurerm_log_analytics_workspace`)  and want to create a new cluster with `azurerm_kubernetes_cluster` and want this new cluster to be

* created with monitoring enabled and
* writing monitor data into the existing workspace.

In the azure portal I have the option to [enable monitoring](https://docs.microsoft.com/en-us/azure/aks/media/container-service-walkthrough-portal/aks-portal-3.png) when creating a new cluster. As I don't find an equivalent in the current [azurerm_kubernetes_cluster](https://www.terraform.io/docs/providers/azurerm/r/kubernetes_cluster.html) I think that it could be a valuable feature to add.
